### PR TITLE
AP_Mount/GCS_MAVLink: move handling of all do-set-roi to AP_Mount library

### PR DESCRIPTION
**Pull request by khanasif786**
In reference to issue #24571 
Initial Draft
(Not tested yet so Draft)

### CODE REVIEW

File: ArduCopter/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -762,7 +762,7 @@ MAV_RESULT GCS_MAVLINK_Copter::handle_command_int_packet(const mavlink_command_i
 }
 
 #if HAL_MOUNT_ENABLED
-MAV_RESULT GCS_MAVLINK_Copter::handle_command_mount(const mavlink_command_long_t &packet, const mavlink_message_t &msg)
+MAV_RESULT GCS_MAVLINK_Copter::handle_command_long_mount(const mavlink_command_long_t &packet, const mavlink_message_t &msg)
```
Comment (peterbarker) [inline ArduCopter/GCS_Mavlink.cpp:765]: This doesn't look right.  We should be accepting *anything* as *just* command_long.

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 2]
```diff
@@ -5128,14 +5100,19 @@ MAV_RESULT GCS_MAVLINK::handle_command_storage_format(const mavlink_command_int_
...
     case MAV_CMD_DO_SET_ROI_SYSID:
-        return handle_command_do_set_roi_sysid(packet);
     case MAV_CMD_DO_SET_ROI_NONE:
-        return handle_command_do_set_roi_none();
+        {
+            AP_Mount *mount = AP_Mount::get_singleton();
+            if (mount == nullptr) {
+                return MAV_RESULT_UNSUPPORTED;
+            }
+            return mount->handle_command_int(packet);
```
Comment (rmackay9) [inline libraries/GCS_MAVLink/GCS_Common.cpp:5114]: We should also send the "mavlink_message_t" as an argument.

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 3]
```diff
@@ -5128,14 +5100,19 @@ MAV_RESULT GCS_MAVLINK::handle_command_storage_format(const mavlink_command_int_
 MAV_RESULT GCS_MAVLINK::handle_command_int_packet(const mavlink_command_int_t &packet)
```
Comment (rmackay9) [inline libraries/GCS_MAVLink/GCS_Common.cpp:5100]: This GCS_MAVLINK::handle_command_int_packet() method is going to need to be changed to include a new argument, "const mavlink_message_t &msg"

**Comment by rmackay9** [comment]
PeterB says, move this block from GCS_MAVLink::handle_command_long() to GCS_MAVLink::handle_command_int()..

   case MAV_CMD_DO_MOUNT_CONFIGURE:
    case MAV_CMD_DO_MOUNT_CONTROL:
    case MAV_CMD_DO_GIMBAL_MANAGER_PITCHYAW:
    case MAV_CMD_DO_GIMBAL_MANAGER_CONFIGURE:
        // some mount commands require the source sysid and compid
        result = handle_command_mount(packet, msg);
        break;

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 4]
```diff
@@ -5128,14 +5100,19 @@ MAV_RESULT GCS_MAVLINK::handle_command_storage_format(const mavlink_command_int_
...
     case MAV_CMD_DO_SET_ROI:
     case MAV_CMD_DO_SET_ROI_LOCATION:
         return handle_command_do_set_roi(packet);
-#if HAL_MOUNT_ENABLED
     case MAV_CMD_DO_SET_ROI_SYSID:
-        return handle_command_do_set_roi_sysid(packet);
     case MAV_CMD_DO_SET_ROI_NONE:
-        return handle_command_do_set_roi_none();
+        {
+            AP_Mount *mount = AP_Mount::get_singleton();
```
Comment (peterbarker) [inline libraries/GCS_MAVLink/GCS_Common.cpp:5110]: All of this should be moved into `void GCS_MAVLINK::handle_command_int(const mavlink_message_t &msg)` instead

**Review by peterbarker — CHANGES_REQUESTED** [review]

**Review by peterbarker — APPROVED** [review]
LGTM

**Comment by rmackay9** [comment]
@khanasif786, thanks for this.  Can you clarify what testing has been done?  .. and maybe update the Description above..

**Comment by rajviCE** [comment]
.

**Pull request merged**